kore/src/Kore/Step/Simplification/TermLike.hs

```diff
@@ -428,8 +425,7 @@ simplify sideCondition = \termLike ->
                 -- TODO(virgil): Move next up through patterns.
                 NextF nextF -> Next.simplify <$> simplifyChildren nextF
                 OrF orF -> Or.simplify <$> simplifyChildren orF
-                RewritesF rewritesF ->
-                    Rewrites.simplify <$> simplifyChildren rewritesF
+                RewritesF _ -> error "Attempting to simplify a Rewrites term"
```

Comment:
We should move this case to the top (next to the `doNotSimplify` ones), and document why it's an error. Also, since this means there's a bug, we should let people know they should report the error to us.